pxr/usd/usdGeom/testenv/testUsdGeomValidators.cpp

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright 2024 Pixar
+//
+// Licensed under the terms set forth in the LICENSE.txt file available at
+// https://openusd.org/license.
+//
+
+#include "pxr/usd/usd/validator.h"
+#include "pxr/usd/usd/validationError.h"
+#include "pxr/usd/usdGeom/validatorTokens.h"
+#include "pxr/usd/usd/validationRegistry.h"
+#include "pxr/usd/usdGeom/metrics.h"
+#include "pxr/usd/usdGeom/tokens.h"
+
+#include <iostream>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+static
+void TestUsdStageMetadata()
+{
+    // Get stageMetadataChecker
+    UsdValidationRegistry &registry = UsdValidationRegistry::GetInstance();
+    const UsdValidator *validator = registry.GetOrLoadValidatorByName(
+            UsdGeomValidatorNameTokens->stageMetadataChecker);
+    TF_AXIOM(validator);
+
+    // Create an empty stage
+    SdfLayerRefPtr rootLayer = SdfLayer::CreateAnonymous();
+    UsdStageRefPtr usdStage = UsdStage::Open(rootLayer);
+
+    UsdValidationErrorVector errors = validator->Validate(usdStage);
+
+    // Verify both metersPerUnit and upAxis errors are present
+    TF_AXIOM(errors.size() == 2);
+    auto rootLayerIdentifier = rootLayer->GetIdentifier().c_str();
+    std::vector<std::string> expectedErrorMessages = {
+            TfStringPrintf("Stage with root layer <%s> does not specify its linear scale in metersPerUnit.", rootLayerIdentifier),
+            TfStringPrintf("Stage with root layer <%s> does not specify an upAxis.", rootLayerIdentifier)
+    };
+
+    for(int i = 0; i < errors.size(); ++i)
```

Comment:
size() returns a size_t (unsigned_int) and the index here is an int. (I corrected this in the internal sync already, just fyi).